### PR TITLE
docs: Add bentoml deployment docs

### PR DESCRIPTION
**Pull request by Sherlock113**
## What does this PR address?

Add docs for `bentoml deployment` in the BentoCloud docs.

Previews:
Screenshot 2023-06-27 at 18 55 19

Screenshot 2023-06-27 at 18 54 51

Screenshot 2023-06-27 at 18 55 03

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [x] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: docs/source/bentocloud/how-tos/deploy-bentos.rst [hunk 1]
```diff
@@ -15,46 +15,152 @@ Online Service
...
+Deploying Your Bento
+--------------------
+Currently, the BentoML CLI only supports creating and updating a Bento deployment by specifying a JSON file, which contains detailed configurations of the deployment, such as ``name``, ``mode``, and ``targets``.
+The JSON file follows the same syntax as the **JSON** tab when you create or update a deployment on the BentoCloud Console.
+
+Run the following command to deploy a Bento.
+
+.. code-block:: bash
+
+   bentoml deployment create --file <file_name>.json
```
Comment (aarnphm) [inline docs/source/bentocloud/how-tos/deploy-bentos.rst:111]: can we also provide a heredoc example here
```bash

bentoml deployment create --file <<EOF
...
EOF
```

File: docs/source/bentocloud/how-tos/deploy-bentos.rst [hunk 2]
```diff
@@ -15,46 +15,152 @@ Online Service
...
+
+.. code-block:: bash
+
+   bentoml deployment get <deployment_name>
+
+Run the following command to update a deployment.
+
+.. code-block:: bash
+
+   bentoml deployment update --file <file_name>.json
```
Comment (aarnphm) [inline docs/source/bentocloud/how-tos/deploy-bentos.rst:133]: Same here. Would be nice if we can make a tab with sphinx-design

**Comment by jianshen92** [comment]
Should we capitalize "D" of the Deployment since its a specific noun in our context?

File: docs/source/bentocloud/how-tos/deploy-bentos.rst (hunk 1 above)
Comment (Haivilo) [reply docs/source/bentocloud/how-tos/deploy-bentos.rst:111]: It seems heredoc cannot be processed now somehow, although it worked before:
```bash
➜  ~ bentoml deployment create --file <<EOF
heredoc> hello
heredoc> EOF
Error: Option '--file' requires an argument.
```
I guess we can leave that part out until I fix it :(

**Review by Haivilo — CHANGES_REQUESTED** [review]
lgtm! I agree we prob should capitalize “D” of Deployment.

File: docs/source/bentocloud/how-tos/deploy-bentos.rst (hunk 1 above)
Comment (Sherlock113) [reply docs/source/bentocloud/how-tos/deploy-bentos.rst]: Thanks. Added.

File: docs/source/bentocloud/how-tos/deploy-bentos.rst (hunk 2 above)
Comment (Sherlock113) [reply docs/source/bentocloud/how-tos/deploy-bentos.rst]: Tab added. Preview:
Screenshot 2023-06-28 at 10 30 24

**Review by Haivilo — APPROVED** [review]

File: docs/source/bentocloud/how-tos/deploy-bentos.rst [hunk 3]
```diff
@@ -7,54 +7,214 @@ Deploying a machine learning model can be quite a task. However, BentoCloud simp
 Deployment Mode
 ==============
```
Comment (aarnphm) [inline docs/source/bentocloud/how-tos/deploy-bentos.rst:8]:
```suggestion
===============
```

**Review by aarnphm — APPROVED** [review]

**Pull request merged**